8bitdo-firmware.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/python3
+
+# based on https://ladis.cloud/blog/posts/firmware-update-8bitdo.html
+
+import requests,json,sys,os
```

Comment:
can you run `black` on this please.

Reply:
Tried that, but it chokes on all the f-strings, even though I used `py312` as target version? Not sure why...